Common/include/containers/CTrapezoidalMap.hpp

```diff
@@ -0,0 +1,107 @@
+/*!
+ * \file TrapezoidalMap.hpp
+ * \brief Implementation of the trapezoidal map for tabulation and lookup of fluid properties
+ * \author D. Mayer, T. Economon
+ * \version 7.3.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2022, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "../../Common/include/linear_algebra/blas_structure.hpp"
+#include "../../Common/include/toolboxes/CSquareMatrixCM.hpp"
+
+/*!
+ * \class CTrapezoidalMap
+ * \brief Construction of trapezoidal map for tabulated lookup 
+ * \author: D. Mayer, T. Economon
+ * \version 7.3.1 "Blackbird"
+ */
+  class CTrapezoidalMap {
+ protected:
+
+  /* The unique values of x which exist in the data */
+  std::vector<su2double> unique_bands_x;
+
+  su2activematrix edge_limits_x;
+  su2activematrix edge_limits_y;
+
+  std::vector<std::vector<unsigned long> > edge_to_triangle;
+
+  /* The value that each edge which intersects the band takes within that
+   * same band. Used to sort the edges */
+  std::vector<std::vector<std::pair<su2double, unsigned long> > > y_edge_at_band_mid;
+
+ public:
+  CTrapezoidalMap(){};
+
+  CTrapezoidalMap(const su2double* samples_x,
+                  const su2double* samples_y,
+                  const unsigned long size,
+                  const std::vector<std::vector<unsigned long> >& edges,
+                  const std::vector<std::vector<unsigned long> >& edge_to_triangle);
+
+  ~CTrapezoidalMap(void){};
+
+  /*!
+   * \brief return the index to the triangle that contains the coordinates (val_x,val_y) 
+   * \param[in]  val_x  - x-coordinate or first independent variable 
+   * \param[in]  val_y  - y-coordinate or second independent variable
+   * \param[out] val_index - index to the triangle 
+   */
+  unsigned long GetTriangle(su2double val_x, su2double val_y);
```

Comment:
Put const on everything please, I have to ask this so many times, and it's so simple... I need to leave SU2.
```suggestion
  unsigned long GetTriangle(su2double val_x, su2double val_y) const;
```

Reply:
GetTriangle calls CTrapezoidalMap::getBand(), which then also has to be made const (the same as for GetTriangle)? But this was not possible due to the variable unique_bands_x used inside this function, from CTrapezoidalMap.hpp:
`std::vector<su2double> unique_bands_x;`
I don't think we can const it. Only thing I managed to do was for GetEdges:
 `CTrapezoidalMap::GetEdges(...) const {}`